FixDevV3Blobs/FixDevV3Blobs/Program.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+
+namespace FixDevV3Blobs
+{
+    class Program
+    {
+        const int MaxTasks = 32;
+
+        const string ProcessedListFilename = "processed.txt";
+        private static HashSet<string> ProcessedBlobs = new HashSet<string>();
+        private static object ProcessedOutputLock = new object();
+
+        static async Task<int> Main(string[] args)
+        {
+            ServicePointManager.DefaultConnectionLimit = MaxTasks * 2 + 1;
+
+            const string connectionString = "<nope>";
+            const string containerName = "v3-catalog0";
+            const string search = "az635243.vo.msecnd.net/v3-catalog0";
+            const string replace = "apidev.nugettest.org/v3/catalog0";
+
+            var account = CloudStorageAccount.Parse(connectionString);
+            var client = account.CreateCloudBlobClient();
+            var container = client.GetContainerReference(containerName);
+
+            LoadProcessedList();
+
+            var blobList = await LoadBlobList(container);
+
+            Log("Got {0} blobs", blobList.Count);
+
+            var stopwatch = await ProcessBlobList(blobList, search, replace);
+
+            Log(
+                "Processed {0} entries in {1}, processing speed: {2} entries per hour, {3} per 1M. Used {4} tasks.",
+                blobList.Count,
+                stopwatch.Elapsed,
+                blobList.Count / stopwatch.Elapsed.TotalHours,
+                TimeSpan.FromTicks(stopwatch.Elapsed.Ticks * (1000000 / blobList.Count)),
+                MaxTasks);
+
+            return 0;
+        }
+
+        private static void LoadProcessedList()
+        {
+            if (File.Exists(ProcessedListFilename))
+            {
+                var lines = File.ReadAllLines(ProcessedListFilename);
+                ProcessedBlobs.UnionWith(lines);
+            }
+        }
+
+        private static async Task<List<CloudBlockBlob>> LoadBlobList(CloudBlobContainer container)
+        {
+            var blobList = new List<CloudBlockBlob>();
+
+            BlobContinuationToken continuationToken = null;
+            var stopwatch = Stopwatch.StartNew();
+            var n = 5;
```

Comment:
Is this used anywhere?

Reply:
Not anymore, was used in the first commit